File: lua/difference-of-squares/difference-of-squares.lua
local diff = {}

function diff.square_of_sum(n)
  local sum = 0

  for i = 1, n, 1 do
    sum = sum + i
  end

  return sum * sum
end

function diff.sum_of_squares(n)
  local sum = 0

  for i = 1, n, 1 do
    sum = sum + (i * i)
  end

  return sum

end

function diff.difference_of_squares(n)
  return diff.square_of_sum(n) - diff.sum_of_squares(n)
end

return diff

File: nim/difference-of-squares/difference_of_squares.nim
proc squareOfSum*(value: int): int =
    result = int(value * (value + 1) / 2)
    result = result * result

proc sumOfSquares*(value: int): int =
    result = int((value * (value + 1) * (2 * value + 1)) / 6)

proc difference*(input: int): int =
    result = squareOfSum(input) - sumOfSquares(input)
